internal/pkg/agent/application/monitoring/v1_monitor_test.go

```diff
@@ -186,11 +186,20 @@ func TestMonitoringConfigComponentFields(t *testing.T) {
 					}
 					continue
 				}
-				if binary != "filebeat" {
-					t.Errorf("expecting fields['binary'] = 'filebeat', got %q", binary)
-				}
-				if componentID != "filestream-default" {
-					t.Errorf("expecting fields['id'] = 'filestream-default', got %q", componentID)
+				if !strings.Contains(componentID, "monitoring") {
+					if binary != "filebeat" {
+						t.Errorf("expecting fields['binary'] = 'filebeat', got %q", binary)
+					}
+					if componentID != "filestream-default" {
+						t.Errorf("expecting fields['id'] = 'filestream-default', got %q", componentID)
+					}
+				} else {
+					if binary != "filebeat" && binary != "metricbeat" {
+						t.Errorf("expected monitoring compoent to be metricbeat or filebeat, got %s", binary)
+					}
+					if componentID != "filestream-monitoring" && componentID != "beat/metrics-monitoring" && componentID != "http/metrics-monitoring" {
```

Comment:
optional.  I'm wondering if using the const `monitoringFilesUnitsID` might be better than the string "filestream-monitoring"